code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/identity/TokenResourceManager.java

```diff
@@ -115,4 +137,35 @@ public void setTokenCreators(List<TokenCreator> tokenCreators) {
         this.tokenCreators = tokenCreators;
     }
 
+    @Override
+    protected Object deleteInternal(String type, String id, Object obj, ApiRequest request) {
+        if (!StringUtils.equals(AbstractTokenUtil.TOKEN, request.getType())) {
+            return null;
+        }
+        return deleteToken(obj, request);
+    }
+
+    protected Object deleteToken(Object obj, ApiRequest request) {
+        Token token = new Token();
+        String jwt = "";
+
+        token = listToken();
+        jwt = token.getJwt();
+
+        if(StringUtils.isBlank(jwt)) {
+            throw new ClientVisibleException(ResponseCodes.INTERNAL_SERVER_ERROR,
+                    "JWTNotProvided", "Request does not contain JWT cookie", null);
+        }
+
+        if (!StringUtils.equals(request.getId(), token.getId())) {
+            throw new ClientVisibleException(ResponseCodes.INTERNAL_SERVER_ERROR,
+                    "IdMismatch", "Id in request does not match token's id", null);
+        }
+
+        request.setResponseCode(ResponseCodes.NO_CONTENT);
+        if(authTokenDao.deleteToken(jwt)) {
+            return obj;
+        }
+        return null;
```

Comment:
@cjellick I didn't get that. Currently the token will get removed from the db when DELETE is called. So do we not want to delete the token and only set it's expiry to Now()/something in past?